lib/consensus/proposer_selector.go

```diff
@@ -17,5 +17,8 @@ type SequentialSelector struct {
 func (s SequentialSelector) Select(blockHeight uint64, roundNumber uint64) string {
 	candidates := sort.StringSlice(s.cm.AllValidators())
 	candidates.Sort()
+	if len(candidates) == 0 {
```

Comment:
How about checking the validators are empty or not ahead of Selecotor, I think `NewNodeRunner` will be good.

Reply:
Very good idea. And I added checking panic statement in NewValidatorConnectionManager.